app/src/main/java/com/github/sdp/tarjetakuna/model/Coordinates.kt

```diff
@@ -14,6 +19,21 @@ data class Coordinates(
         longitude = longitude.coerceAtMost(longitudeRange).coerceAtLeast(-longitudeRange)
     }
 
+    /**
+     * Calculates the distance in kilometers to another [Coordinates].
+     */
+    fun distanceKmTo(other: Coordinates): Float {
+        // Haversine formula
+        val earthRadius = 6371
+        val latDistance = Math.toRadians((latitude - other.latitude).toDouble())
+        val lngDistance = Math.toRadians((longitude - other.longitude).toDouble())
+        val a = (sin(latDistance / 2) * sin(latDistance / 2)
```

Comment:
Consider more descriptive val names

Reply:
I based my variables off the Haversine formula available [here](https://en.wikipedia.org/wiki/Haversine_formula).
a and c are therefore used as is.

Reply:
Ok, consider mentioning this in a comment just above. But I would not keep the names a and c, it took me a couple seconds to map exactly what you were talking about to the wikipedia page because they don't use a and c in that way. You can write 
 val haversineLatitude = (sin(latDistance / 2) * sin(latDistance / 2)
 val haversineLongitude = (sin(latDistance / 2) * sin(latDistance / 2)
 val haversine = havLatitude + cos(latitude)cos(other.latitude)*havLongitude
 I am still not sure where you got c from, the closest thing to your formual they call d (the distance) but then they use 
 distance = 2*earthradius*arcsin(sqrt(haversine))

Reply:
I'm sorry, but I don't think it is very useful for this function. I won't change it.

Reply:
Your function makes no sense as is.